### PR TITLE
TW-1936: fix the share screen don't show up when sharing an image

### PR DESCRIPTION
**Pull request by sherlockvn**
## Ticket
**Related issue**
#1936 
## Solution
**Outline the implemented solution, detailing the changes made and how they address the issue**
Change from `go` to `push`

## Resolved
**Attach screenshots or videos demonstrating the changes**
- Web:
- Android:

https://github.com/user-attachments/assets/f98fb53c-e0f5-49da-94d6-edbc605fbea3

- IOS:

https://github.com/user-attachments/assets/97314f25-234b-434c-9977-cebcb064aa95

### CODE REVIEW

**Comment by nqhhdev** [comment]
I found an issue for this PR, reproduce steps:

- Open the app and go to the chat list
- Open a chat and go to chat profile
- Leave the app and share something (File, image, text)
- Received share and open share screen
- Send to another chat.
- Can't open exactly chat and can't back to previous screen

https://github.com/user-attachments/assets/923a83c1-9c6a-437a-96f5-35e8a79d8c1e

**Comment by sherlockvn** [comment]
> I found an issue for this PR, reproduce steps:
> 
> * Open the app and go to the chat list
> * Open a chat and go to chat profile
> * Leave the app and share something (File, image, text)
> * Received share and open share screen
> * Send to another chat.
> * Can't open exactly chat and can't back to previous screen
> 
>  Simulator.Screen.Recording.-.iPhone.15.-.2024-08-08.at.17.13.24.mp4

https://github.com/user-attachments/assets/c40de08e-5907-4ed5-9d2b-a7d3f572dc3f

**Comment by Te-Z** [comment]
Found an other issue on android:

- start typing something in search text field
- leave the app and share something
- the file is successfully shared
- we now have two instances of twake chat (one with the sharing, the other one looks like the search screen)
- when trying to go back on the first one there's a black screen, the only solution is to kill the app

Expectation: when a file is shared only one instance of the app appear and there's no black screen

https://github.com/user-attachments/assets/4a92c71f-f9a5-4db3-9c39-0b907d5a03a7

**Comment by sherlockvn** [comment]
It doesn't happend to me. Can you delete all twake chat app first before test ?
https://github.com/user-attachments/assets/1f0e8680-9048-45c2-9241-fbf13b6fe191



> Found an other issue on android:
> 
> * start typing something in search text field
> * leave the app and share something
> * the file is successfully shared
> * we now have two instances of twake chat (one with the sharing, the other one looks like the search screen)
> * when trying to go back on the first one there's a black screen, the only solution is to kill the app
> 
> Expectation: when a file is shared only one instance of the app appear and there's no black screen
> 
>  screen-20240820-072605.mp4

File: lib/pages/chat_list/receive_sharing_intent_mixin.dart [hunk 1]
```diff
@@ -35,7 +36,14 @@ mixin ReceiveSharingIntentMixin<T extends StatefulWidget> on State<T> {
...
         )
         .toList();
-    TwakeApp.router.go('/share');
+    openSharePage();
+  }
+
+  void openSharePage() {
+    if (TwakeApp.router.activeRoomId?.isNotEmpty == true) {
+      TwakeApp.router.go('/rooms');
+    }
```
Comment (nqhhdev) [inline lib/pages/chat_list/receive_sharing_intent_mixin.dart:45]:
```suggestion
    if (TwakeApp.router.routeInformationProvider.value.uri.path.startsWith('/rooms/')) {
      TwakeApp.router.go('/rooms');
    }
```

**Comment by nqhhdev** [comment]
Failed for this case:
- Open Settings 
- Open account profile
- Click on `Switch accounts` or `Add another account` → open bottom sheet
- leave the app
- Share something for Twake Chat
- Open share screen however can't close bottom sheet

File: lib/pages/chat_list/receive_sharing_intent_mixin.dart [hunk 2]
```diff
@@ -35,9 +36,32 @@ mixin ReceiveSharingIntentMixin<T extends StatefulWidget> on State<T> {
...
+    );
+  }
+
+  bool isCurrentPageIsInRooms() =>
+      TwakeApp.router.routeInformationProvider.value.uri.path
+          .startsWith('/rooms/');
+
+  bool isCurrentPageIsNotRooms() =>
+      !TwakeApp.router.routeInformationProvider.value.uri.path
+          .startsWith('/rooms');
```
Comment (hoangdat) [inline lib/pages/chat_list/receive_sharing_intent_mixin.dart:63]: please test with:
- tablet
- web app

Comment (nqhhdev) [inline lib/pages/chat_list/receive_sharing_intent_mixin.dart:63]:
```
  bool isCurrentPageIsNotRooms() =>
      !TwakeApp.router.routeInformationProvider.value.uri.path
          .startsWith('/rooms/');
```

What is the different for this?

Comment (sherlockvn) [reply lib/pages/chat_list/receive_sharing_intent_mixin.dart:63]: All pages that are not start with `/rooms`, for example: `/rooms`, and `/twakeWelcome` will have isCurrentPageIsNotRooms is true

**Comment by nqhhdev** [comment]
Hi @sherlockvn I found an issue on Mobile (iOS and Android) related to your PR.


https://github.com/user-attachments/assets/4243463b-64c7-4ed8-b728-4b076a2440b3


https://github.com/user-attachments/assets/788a1aa5-1b76-4d75-803a-cd384a32e6e6

**Comment by sherlockvn** [comment]
> Hi @sherlockvn I found an issue on Mobile (iOS and Android) related to your PR.
> 
>  Screenrecorder-20240906-103340.mp4 
>  Simulator.Screen.Recording.-.iPhone.15.-.2024-09-06.at.10.23.06.mp4 
> ### Tested on main branch
>  Simulator.Screen.Recording.-.iPhone.15.-.2024-09-06.at.10.42.35.mp4


https://github.com/user-attachments/assets/04e898dd-4bf9-49d8-ad2a-4906dfb7c2b1

**Review by nqhhdev — APPROVED** [review]
Tested on iOS and Android, works well for me.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**